service/storage_service.cc

```diff
@@ -2985,7 +2985,12 @@ future<> storage_service::restore_replica_count(inet_address endpoint, inet_addr
             slogger.warn("Streaming to restore replica count failed: {}", std::current_exception());
             // We still want to send the notification
     }
-    this->send_replication_notification(notify_endpoint).get();
+    try {
+        this->send_replication_notification(notify_endpoint).get();
+    } catch (...) {
+        slogger.warn("Sending replication notification to {} failed: {}", notify_endpoint, std::current_exception());
+        // We still want to stop the status checker
```

Comment:
What's the implication of ignoring the error? Should we preserve it for later re-throw?

If it's okay to swallow the exception, a comment should explain why.

Reply:
To preserve the existing behavior we indeed need to rethrow the exception in this case.
But since the only thing we do with this error is print another warning in https://github.com/scylladb/scylladb/blob/63c2cdafe88aea439a8a8160674bd75af72471a0/service/storage_service.cc#L1119-L1121
it's ok to swallow it here.

Reply:
Ok. But anything that swallows an exception needs a comment to explain why it's okay.